### PR TITLE
fix(store): make executeQuery and objectMatchesQuery agree on matcher semantics and empty expressions

### PR DESCRIPTION
**Pull request by steveruizok**
**Risk: medium · Importance: medium**

This PR fixes a bug where a store query could return different results depending on whether it was computed from scratch (via the indexes) or updated incrementally (via the predicate). Split out of #10177.

### Before

`executeQuery` (index lookup, used from scratch) and `objectMatchesQuery` (predicate, used for incremental updates) each had their own matching logic, and they disagreed:

- `ids('book', () => ({ tag: { eq: undefined } }))` was 0 from scratch (the indexes only track defined values) but 1 after an incremental add (the predicate compared `undefined === undefined`).
- `{ eq: NaN }` matched via the index (a `Map` lookup uses SameValueZero) but never via the predicate (`!==`).
- A matcher with several operators such as `{ gt: 70, neq: 75 }` only honoured the first operator in the index path (`if ('eq' in matcher) ... else if ('neq' in matcher) ... else if ('gt' in matcher)`), while the predicate applied all of them.
- `exec('book', {})` returned `[]` with records present: `executeQuery` intersected over zero matcher paths, and only `StoreQueries.ids` special-cased the empty expression with `hasAnyKey`.

### After

`executeQuery.ts` has one `matchesValue(matcher, value)` predicate using SameValueZero that both strategies share: an `undefined` value matches no matcher, and all operators in a matcher apply. `executeQuery` uses a direct index lookup only for a lone `eq` and otherwise scans the index's values through `matchesValue`; with no matcher paths it returns `store.getAllIdsForType(typeName)`. `objectMatchesQuery` treats an empty nested sub-expression (or a non-object entry) as no constraint, mirroring `extractMatcherPaths`. `StoreQueries.ids` drops its `hasAnyKey` special case and passes the query through to `fromScratch`/`fromScratchWithDiff`. `packages/store/SPEC.md` QE1, QE3, QE4 and QQ5 are updated.

### API changes

The stray duplicate `/** @public */` above `QueryExpression` is removed, so `QueryExpression` goes from `@public (undocumented)` to `@public` (documented) in `api-report.api.md`. No runtime API change.

### Change type

- [x] `bugfix`

### Test plan

- [x] Unit tests — the new `[QE1]`, `[QE3]`, `[QE4]`, `[QQ5]` tests (and the updated `should handle empty query` test) fail on `main` and pass with this change

### Release notes

- Fix `store.query` matching: `{ eq: undefined }` no longer matches incrementally, `NaN` matches by SameValueZero in both strategies, matchers with several operators apply all of them, and `exec(type, {})` returns every record of the type

### Code changes

| Section         | LOC change |
| --------------- | ---------- |
| Core code       | +40 / -33  |
| Tests           | +59 / -2   |
| Documentation   | +4 / -4    |
| Automated files | +1 / -1    |

### CODE REVIEW

File: packages/store/src/lib/executeQuery.ts [hunk 1]
```diff
@@ -82,23 +82,34 @@ function extractMatcherPaths(
...
+
+// The one place matcher semantics live. `objectMatchesQuery` (the predicate used for incremental
+// updates) and `executeQuery` (the index lookup used from scratch) must agree, and the indexes only
+// track defined values, so an undefined value never matches any matcher.
+function matchesValue(matcher: QueryValueMatcher<any>, value: unknown): boolean {
+	if (value === undefined) return false
+	if ('eq' in matcher && !sameValueZero(value, matcher.eq)) return false
+	if ('neq' in matcher && sameValueZero(value, matcher.neq)) return false
+	if ('gt' in matcher && (typeof value !== 'number' || value <= matcher.gt)) return false
+	return true
```
[the remark this answers is not in the document]
Comment (steveruizok) [reply packages/store/src/lib/executeQuery.ts:99]: Good catch. The rewrite turned `value > gt` into `!(value <= gt)`, which flips for `NaN` (both as the record value and as the bound). Fixed in ddef4b5d7d: `matchesValue` now uses `!(typeof value === 'number' && value > matcher.gt)`, and there is a new `[QE1]` agreement test in `executeQuery.test.ts` that checks `{ gt: 0 }` excludes a `NaN` value and `{ gt: NaN }` matches nothing in both the index and predicate paths.

**Comment by MitjaBezensek** [comment]
Closing this. The shared `matchesValue` fixes the eq undefined, NaN, multi-operator and empty-expression cases, but the index and predicate paths still disagree on nested sub-expressions with no matcher below depth 1: `{ metadata: { extras: {} } }` or `{ metadata: { x: 5 } }` returns every record from scratch (`extractMatcherPaths` finds nothing), while `objectMatchesQuery` still requires `metadata` to be an object, because the new `hasAnyKey` check only looks one level down. So reactive `ids()` membership still depends on edit history, and QE3 doesn't hold beyond depth 1.

Tracking the full set of cases in #10919, including what a follow-up fix should cover (a recursive has-any-matcher check, a depth-2 test with the nested field missing or null, and a test through reactive `ids()`).
